src/examples/hotgym/HelloSPTP.cpp

```diff
@@ -36,12 +36,21 @@ namespace examples {
 using namespace std;
 using namespace htm;
 
+/**
+ *  helper to transform (Real) data to categories (UInt) for Classifier/Predictor
+ **/
+UInt realToCategory_(const Real r) {
+  return static_cast<UInt>((r+1.0f /*map sin(x):[-1,1] map it to [0,2]*/)*1000); //precision on 3 dec places
```

Comment:
the `std::bad_alloc` bug (only on Release) was caused here, as the mapping to labels used to underflow (UInt -1), resulting in a huge label, which led to a huge PDF vectors. (not in Debug, because that runs only like 2 steps).